plugins/pointstocsv/pointstocsv.cpp

```diff
@@ -0,0 +1,282 @@
+/*****************************************************************************/
+/*  pointstocsv.cpp - Exports coordinates of points of lines, polylines to   */
+/*  csv file, and as well the coordinates of selected points                 */
+/*                                                                           */
+/*                                                                           */
+/*  Copyright (C) 2023 Joaquin, joaquinperezvalera@gmail.com                 */
+/*                                                                           */
+/*  This library is free software, licensed under the terms of the GNU       */
+/*  General Public License as published by the Free Software Foundation,     */
+/*  either version 2 of the License, or (at your option) any later version.  */
+/*  You should have received a copy of the GNU General Public License        */
+/*  along with this program.  If not, see <http://www.gnu.org/licenses/>.    */
+/*****************************************************************************/
+
+
+#include <QTextEdit>
+#include <QDialogButtonBox>
+#include <QVBoxLayout>
+#include <QPushButton>
+#include <QComboBox>
+#include <QLabel>
+#include <QRect>
+#include <QFileDialog>
+#include <QMessageBox>
+#include <cmath>
```

Comment:
On header order:

we mostly keep c++ standard headers before Qt headers. Once there were some Qt bugs which were triggered if Qt headers are included before c++ standard headers, but in most time, the order is not an issue.